File: Unit Testing/LCA.java
public class LCA {

    static Node lca(Node root, int num0, int num1) {
        while (root != null) {

            if (root.data > num0 && root.data > num1)
                root = root.left;

            else if (root.data < num0 && root.data < num1)
                root = root.right;

            else break;
        }
        return root;
    }
}

File: Unit Testing/LCA.py
class Node:

    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None


def lca(root, num0, num1):
    while root:
        if root.data > num0 and root.data > num1:
            root = root.left

        elif root.data < num0 and root.data < num1:
            root = root.right

        else:
            break

    return root